pester.go

```diff
@@ -165,6 +171,14 @@ func LinearJitterBackoff(i int) time.Duration {
 	return jitter(i)
 }
 
+// DefaultRetryStrategy returns true in the following case:
+//  - status code is lower than http.StatusInternalServerError and
+//  - status is not http.StatusTooManyRequests or status is http.StatusTooManyRequests with retryOnHTTP429 flag set false
+func DefaultRetryStrategy(status int, retryOnHTTP429 bool) bool {
+	return status < http.StatusInternalServerError &&
+		(status != http.StatusTooManyRequests || (status == http.StatusTooManyRequests && !retryOnHTTP429))
+}
```

Comment:
I found this function a bit hard to read easily, so what do you think about switching to simple if cases?

```suggestion
// DefaultRetryStrategy returns true in the following case: // - status code is lower than http.StatusInternalServerError and status is not http.StatusTooManyRequests // - status code is lower than http.StatusInternalServerError and status is http.StatusTooManyRequests // with retryOnHTTP429 flag set to false func DefaultRetryStrategy(status int, retryOnHTTP429 bool) bool { if status < http.StatusInternalServerError && status != http.StatusTooManyRequests { return true } if status < http.StatusInternalServerError && status == http.StatusTooManyRequests && !retryOnHTTP429 { return true } return false }
```